### PR TITLE
send device model in handshake and error reports

### PR DESCRIPTION
**Pull request by mattklein123**
Also bump shared core to bring in previous disconnection reason change in API.

Fixes BIT-4172

### CODE REVIEW

File: platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt [hunk 1]
```diff
@@ -172,6 +173,7 @@ internal class LoggerImpl(
                 eventsListenerTarget,
                 clientAttributes.appId,
                 clientAttributes.appVersion,
+                Build.MODEL,
```
Comment (murki) [inline platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt:176]: This is already abstracted via `DeviceAttributes` we should expose it there

File: platform/swift/source/Logger.swift [hunk 2]
```diff
@@ -128,6 +128,7 @@ public final class Logger {
         let appStateAttributes = AppStateAttributes()
         let clientAttributes = ClientAttributes()
         let deviceAttributes = DeviceAttributes()
+        let hardwareVersion = deviceAttributes.hardwareVersion
```
Comment (murki) [inline platform/swift/source/Logger.swift:131]: is this allocation necessary? seems like we can call `model: deviceAttributes.hardwareVersion` below

Comment (mattklein123) [reply platform/swift/source/Logger.swift:131]: I'm not very familiar with how swift works and what allocates or not. I will read more and clean it up.

File: platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt (hunk 1 above)
Comment (mattklein123) [reply platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt:176]: OK I will change this around.

**Review by murki — APPROVED** [review]
couple of nits, looks fine otherwise

**Pull request merged**